src/NServiceBus.Transport.SqlServer.AcceptanceTests/MultiCatalog/SqlUtilities.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace NServiceBus.Transport.SqlServer.AcceptanceTests.MultiCatalog
+{
+#if SYSTEMDATASQLCLIENT
+    using System.Data.SqlClient;
+#else
+    using Microsoft.Data.SqlClient;
+#endif
+    using System;
+    using System.Threading.Tasks;
+    using System.Threading;
+
+    public static class SqlUtilities
+    {
+        public static async Task<bool> CheckIfTableExists(string catalogName, string schemaName, string tableName, SqlConnection connection, CancellationToken cancellationToken = default)
+            => await RunCommand(connection, async (command, token) =>
```

Comment:
Avoid an async/await state machine by directly returning the task
```suggestion
        public static Task<bool> CheckIfTableExists(string catalogName, string schemaName, string tableName, SqlConnection connection, CancellationToken cancellationToken = default)
            => RunCommand(connection, async (command, token) =>
```

Same in DropTable below.